packages/core/tracing/src/components/waterfall/WaterfallSpanRow.vue

```diff
@@ -0,0 +1,306 @@
+<template>
+  <div
+    :class="['waterfall-row', 'waterfall-span-row', { selected }]"
+    @click="handleSelect"
+  >
+    <div class="label">
+      <template v-if="depth > 1">
+        <WaterfallSpacer
```

Comment:
would you consider reducing the number of DOM nodes by changing the width of the spacer instead? although that would work as well

Reply:
The spacer component not only takes up the space for indentation but also draws the rulers:

<img width="272" alt="Screenshot 2024-11-20 at 16 25 32" src="https://github.com/user-attachments/assets/0bde7bd2-a97c-4413-bc17-4fe02504287d">

Do you feel it is better to draw these ruler segments with paths in an SVG?

Reply:
I will try some CSS tricks like `repeating-linear-gradient` to reduce the count of DOM nodes